lib/GADS.pm

```diff
@@ -2753,6 +2753,235 @@ prefix '/:layout_name' => sub {
+                    {
+                        name        => $report_name,
+                        description => $report_description,
+                        layouts     => $checkboxes
+                    }
+                );
+
+                my $lo = param 'layout_name';
+                return forwardHome( { success => "Report updated" },
+                    "$lo/report" );
+            }
+
+            my $user   = logged_in_user;
+            my $layout = var('layout') or pass;
+
+            if ( app->has_hook('plugin.linkspace.data_before_request') ) {
+                app->execute_hook( 'plugin.linkspace.data_before_request',
+                    user => $user );
+            }
+
+            my %params = (
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $fields = [ $layout->all( user_can_read => 1 ) ];
+
+            my $base_url = request->base;
+
+            my $params;
+
+            $params->{header_type} = 'table_tabs';
+
+            $params->{layout_obj} = $layout;
+            $params->{layout}     = $layout;
+
+            $params->{header_back_url} = "${base_url}table";
+            $params->{breadcrumbs}     = [
+                Crumb( $base_url . "table/", "Tables" ),
+                Crumb( "",                   "Table: " . $layout->name )
+            ];
+
+            my $report_id = param('id');
+
+            my $result = schema->resultset('Report')->load_for_edit($report_id);
+
+            my $report_layouts = $result->report_layouts;
+
+            my $layouts = [];
+
+            while ( my $report_layout = $report_layouts->next ) {
+                push @$layouts, $report_layout->layout_id;
+            }
+
+            foreach my $field (@$fields) {
+                if ( grep { $_ eq $field->{id} } @$layouts ) {
+                    $field->{is_checked} = 1;
+                }
+            }
+
+            $params->{report}   = $result;
+            $params->{fields}   = $fields;
+            $params->{viewtype} = 'edit';
+
+            template 'reports/edit' => $params;
+        };
+
+        #Delete a report (by :id)
+        get "/delete:id" => sub {
+            my $user   = logged_in_user;
+            my $layout = var('layout') or pass;
+
+            my $report_id = param('id');
+
+            my $result = schema->resultset('Report')->load_for_edit($report_id);
+
+            $result->delete;
+
+            my $lo = param 'layout_name';
+            return forwardHome( { success => "Report deleted" }, "$lo/report" );
+        };
+
+        #Render the report (by :report) with the view (by :view)
+        get "/render:report/:view" => sub {
+            my $user = logged_in_user;
+
+            my $report_id = param('report');
+            my $view_id   = param('view');
+
+            my $report =
+              schema->resultset('Report')->load( $report_id, $view_id );
```

Comment:
Thinking about this a bit more, I think a nicer way of implementing it would be to remove `load()` to `Result::Report` and then use `find`. E.g. `my $report = schema->resultset('Report')->find($report_id)->load($view_id)`